StPicoDstMaker/StPicoDstMaker.cxx

```diff
@@ -199,23 +199,36 @@ Int_t StPicoDstMaker::Init(){
     openRead();     // if read
   } else if (mIoMode == ioWrite) {
     openWrite();
-    if(!initMtd()) {
+    if(mEmcMode) initEmc();
+  }
+  return kStOK;
+}
+
+//-----------------------------------------------------------------------
+Int_t StPicoDstMaker::InitRun(const Int_t runnumber){
+  // Dec. 1st is the assumed to the start a new running year
+  mYear = runnumber/1e6 + 1999;
+  if((runnumber%1000)/1000>334) mYear += 1;
+
+  printf("[i]Run = %d, year = %d\n",runnumber,mYear);
```

Comment:
You rely on `StMaker::InitRun` to initialize `mYear`. That run number is fetched from `StEvtHddr` which is only present in the write mode of picoDst. It is empty otherwise. That causes `StPicoDstMaker::InitRun` to print Run = -1, year = 1998 in read mode. It might be confusing for users. I suggest to move the mYear calculation to calculation to inside the the `if(mIoMode == ioWrite)` statement.

If not for that fact that you are using `mYear` in `::fillMtdTrigger` , I would have preferred to keep that calculation local to `::InitRun()` instead of being global to the maker. Just to avoid that someone uses it it in read mode. In any case, it is not a big issue.

Can you also use the logger to print the run and year? Let us avoid c-style printf.